Dining/models.py

```diff
@@ -271,8 +273,11 @@ class DiningEntryUser(DiningEntry, DiningWork):
     def clean(self):
         super().clean()
         if not self.pk:
-            if DiningEntryUser.objects.filter(dining_list=self.dining_list, user=self.user):
-                raise ValidationError(_("User is already on this dining list."))
+            try:
+                if DiningEntryUser.objects.filter(dining_list=self.dining_list, user=self.user):
+                    raise ValidationError(_("User is already on this dining list."))
+            except ObjectDoesNotExist:
```

Comment:
Where can this exception be raised in the try block?

Reply:
Due to the clean_dining_list method in the create form it could happen that the cleaning of dining_list fails (list closed, full, whatever). As a result, the code calling clean_dining_list is returned an error instead of a value (setting self.dininglist to None). However, django still runs the models clean method afterwards for some reason. Thus in the filter query a query with dininglist=None is executed returning an objectDoesNotExist. I don't know of a better solution than to remove this clean method here entirely and moving it to the form instead (though arguably we don't want any case of a duplicate dining entry in the list).
The phrasing in the validationerror was incorrect though, that has been fixed.